docker/docker-compose.yml

```
@@ -1,61 +1,74 @@
 version: '3.7'
+
+# keep command outside anchor because of performance problems
+x-redis: &default-redis
+  image: redis:${REDIS_VERSION:-6.2.5-alpine}
+  deploy:
+    restart_policy:
+      condition: any
+  volumes:
+    - ${APP_HOME:-..}/redis:/data
+
+x-environment: &default-environment
+  PASSENGER_APP_ENV: test
+  REDIS_COUNT_HOST: redis-count
+  REDIS_HOST: redis-cache
+  REDIS_RESQUE_HOST: redis
+
+x-environment-resque: &environment-resque
+  <<: *default-environment
+  APP_ENV: test
+  ROUTER_API_KEY: ${ROUTER_API_KEY:-demo}
+  ROUTER_URL: ${ROUTER_URL:-http://localhost:4899/0.1}
+
+x-default: &default-service
```

Can we use a uniform naming convention?
```suggestion
x-environment-resque: &default-environment-resque
  <<: *default-environment
  APP_ENV: test
  ROUTER_API_KEY: ${ROUTER_API_KEY:-demo}
  ROUTER_URL: ${ROUTER_URL:-http://localhost:4899/0.1}

x-service: &default-service
```

Just to make sure this is voluntary, it looks like the anchor `x-environment-resque` is never referenced, do we need it?

---

You are write, something is wrong because of my going back and forth.

---

Now I understand the naming. I  thought you had a specific naming convention (that adds the "default" to the name) but apparently that is not the case. Than I suggest being verbose and naming the anchors accordingly.That is instead of

```
x-redis: &default-redis

x-environment: &default-environment

x-environment-resque: &environment-resque

x-default: &default-service
```
more like this
```
x-default-redis: &default-redis

x-default-environment: &default-environment

x-resque-environment: &resque-environment

x-default-service: &default-service
```

At the end of the day, if you fixed the other issue, this naming issue doesn't matter that much but it is better to have a coherent naming convention to not to get lost after some time has passed.